app/assets/javascripts/admin/auth/registration_view.js

```diff
@@ -9,24 +9,51 @@ import RegistrationForm from "./registration_form";
 
 type Props = {
   history: RouterHistory,
+  organizationName: ?string,
 };
 
 class RegistrationView extends React.PureComponent<Props> {
+  getGreetingCard() {
+    const { organizationName } = this.props;
+    if (organizationName) {
+      return (
+        <Card style={{ marginBottom: 24 }}>
+          You were invited to join the organization &ldquo;{organizationName}&rdquo;!<br /> In case
+          you do not know this organization, contact{" "}
+          <a href="mailto:hello@scalableminds.com">hello@scalableminds.com</a> to get more
+          information about how to get to use webKnossos.
+        </Card>
+      );
+    }
+
+    return (
+      <Card style={{ marginBottom: 24 }}>
+        Not a member of the listed organizations?<br /> Contact{" "}
+        <a href="mailto:hello@scalableminds.com">hello@scalableminds.com</a> to get more information
+        about how to get to use webKnossos.
+      </Card>
+    );
+  }
+
   render() {
     return (
       <Row type="flex" justify="center" style={{ padding: 50 }} align="middle">
         <Col span={8}>
           <h3>Registration</h3>
-          <Card style={{ marginBottom: 24 }}>
-            Not a member of the listed organizations?<br /> Contact{" "}
-            <a href="mailto:hello@scalableminds.com">hello@scalableminds.com</a> to get more
-            information about how to get to use webKnossos.
-          </Card>
+          {this.getGreetingCard()}
           <RegistrationForm
+            // The key is used to enforce a remount in case the organizationName changes.
+            // That way, we ensure that the organization field is cleared.
+            key={this.props.organizationName || "default registration form key"}
+            organizationName={this.props.organizationName}
             onRegistered={() => {
               Toast.success(messages["auth.account_created"]);
               this.props.history.push("/auth/login");
             }}
+            onOrganizationNameNotFound={() => {
+              Toast.error(messages["auth.invalid_organization_name"]);
+              this.props.history.push("/auth/register");
```

Comment:
In my understanding this error can only be triggered if the user already is on the register page. So I'd say there's no need to push this to the history.

Reply:
Hm, I'm not sure I get what you mean. If this error happens, the user is on `/auth/register?organizationName=someInvalidOrgName`. Then, he should be redirected to ``/auth/register?organizationName`, so that the dropdown appears again. Or do you mean that something like `history.replace` should be used?

Reply:
No, you're right. I didn't get that this line was being used to "get rid" of the `organizationName` url parameter :)